nodestream/extractors/stores/aws/s3_extractor.py

```diff
@@ -55,10 +70,17 @@ def find_keys_in_bucket(self) -> list[str]:
         paginator = self.s3_client.get_paginator("list_objects_v2")
         page_iterator = paginator.paginate(Bucket=self.bucket, Prefix=self.prefix)
         for page in page_iterator:
-            for obj in page["Contents"]:
-                yield obj["Key"]
+            if page["KeyCount"] > 0:
+                for obj in page["Contents"]:
+                    if self.archive_dir is not None:
+                        if obj["Key"].startswith(self.archive_dir):
+                            continue
+                    yield obj["Key"]
 
     async def extract_records(self) -> AsyncGenerator[Any, Any]:
         for key in self.find_keys_in_bucket():
             for record in self.get_object_as_file(key).read_file():
-                yield record
+                try:
+                    yield record
+                finally:
+                    self.archive_s3_object(key)
```

Comment:
```python
for key in self.find_keys_in_bucket():
    try:
        for record in self.get_object_as_file(key).read_file():
            yield record
    finally:
        self.archive_s3_object(key)
```

I think we need to move the `try-finally` block out of the loop so it doesn't try to archive on every record in the file.

Reply:
DOH... i think i fixed it now